### PR TITLE
feat: redesign 'add text' modal

### PR DESCRIPTION
**Pull request by ztlee042**
#1096 

2024-03-28 
CI tests failed. Maybe due to SDK changes.

### CODE REVIEW

**Review by spaenleh — CHANGES_REQUESTED** [review]
Looks good, I have a few comments, but overall I think it is very good !

File: src/components/item/form/DocumentForm.tsx [hunk 1]
```diff
@@ -68,6 +69,22 @@ export const DocumentExtraForm = ({
...
+      case DocumentItemExtraFlavor.Error:
+        return '#FDEDED';
+      case DocumentItemExtraFlavor.Success:
+        return '#EDF7ED';
+      case DocumentItemExtraFlavor.Warning:
+        return '#FFF4E5';
+      default:
+        return null;
+    }
+  };
```
[the remark this answers is not in the document]
Comment (ztlee042) [reply src/components/item/form/DocumentForm.tsx:86]: Yes, it’s a bit different. The colors from palette are bit too dark as background colors. So I get these colors from the design file.

**Review by spaenleh — APPROVED** [review]
Changes look good as far as I am concerned. I think we need to merge the fonts and typography variants before this PR.

File: src/langs/en.json [hunk 2]
```diff
@@ -22,11 +22,14 @@
   "CREATE_ITEM_NEW_FOLDER_TITLE": "Create a Folder",
   "CREATE_NEW_ITEM_APP_TITLE": "Create an App",
   "CREATE_NEW_ITEM_APP_URL_LABEL": "App URL",
-  "CREATE_NEW_ITEM_DOCUMENT_TITLE": "Create a Document",
+  "CREATE_NEW_ITEM_DOCUMENT_TITLE": "Add Text",
   "CREATE_NEW_ITEM_ETHERPAD_INFORMATIONS": "Etherpad is a collaborative real-time text editor",
   "CREATE_NEW_ITEM_ETHERPAD_LABEL": "Name",
   "CREATE_NEW_ITEM_ETHERPAD_TITLE": "Create an Etherpad",
   "CREATE_NEW_ITEM_NAME_LABEL": "Name",
+  "CREATE_NEW_ITEM_NAME_HELPER_TEXT": "Show in the builder",
```
Comment (pyphilia) [inline src/langs/en.json:30]:
```suggestion
  "CREATE_NEW_ITEM_NAME_HELPER_TEXT": "Show in Graasp Builder",
```

File: src/langs/en.json [hunk 3]
```diff
@@ -22,11 +22,14 @@
...
+  "CREATE_NEW_ITEM_DISPLAY_NAME_LABEL": "Display Name",
+  "CREATE_NEW_ITEM_DISPLAY_NAME_HELPER_TEXT": "Show in the player. Defaults to the 'Name' field. Customize it to your needs.",
```
Comment (pyphilia) [inline src/langs/en.json:32]:
```suggestion
  "CREATE_NEW_ITEM_DISPLAY_NAME_HELPER_TEXT": "Show in Graasp Player. Defaults to the 'Name' field. Customize it to your needs.",
```

File: src/components/main/NewItemModal.tsx [hunk 4]
```diff
@@ -182,7 +182,7 @@ const NewItemModal = ({ open, handleClose }: Props): JSX.Element => {
       case ItemType.DOCUMENT:
         return (
           <>
-            <Typography variant="h6">
+            <Typography variant="h2" color="primary">
```
Comment (pyphilia) [inline src/components/main/NewItemModal.tsx:185]: I would let it untouched, or you change all the others. But in any case I think h2 is way too big.

**Review by pyphilia — APPROVED** [review]
Thanks for this PR! I've left some suggestions, let me know if you have questions.

**Comment by ztlee042** [comment]
CleanShot 2024-04-04 at 14 20 25
# Fix Select

- Add icons
- Apply colors via theme 
- Select 'standard' by default

# Future work
- Change the hover color on the selected item, current one is a purple-ish color, which is not consistent with the gray-ish color on unselected items
- Align the select, menu items with the text area

**Comment by ztlee042** [comment]
# Add Tooltip

- move helper text to tooltip
- remove the built-in required feature since it takes another row instead on the same row with the new label
- add a custom required start
CleanShot 2024-04-04 at 14 53 45

File: src/langs/en.json [hunk 5]
```diff
@@ -22,11 +22,14 @@
...
   "CREATE_NEW_ITEM_APP_URL_LABEL": "App URL",
-  "CREATE_NEW_ITEM_DOCUMENT_TITLE": "Create a Document",
+  "CREATE_NEW_ITEM_DOCUMENT_TITLE": "Add Text",
   "CREATE_NEW_ITEM_ETHERPAD_INFORMATIONS": "Etherpad is a collaborative real-time text editor",
   "CREATE_NEW_ITEM_ETHERPAD_LABEL": "Name",
   "CREATE_NEW_ITEM_ETHERPAD_TITLE": "Create an Etherpad",
   "CREATE_NEW_ITEM_NAME_LABEL": "Name",
+  "CREATE_NEW_ITEM_NAME_HELPER_TEXT": "Show in Graasp builder",
+  "CREATE_NEW_ITEM_DISPLAY_NAME_LABEL": "Display Name",
+  "CREATE_NEW_ITEM_DISPLAY_NAME_HELPER_TEXT": "Show in Graasp player. Defaults to the 'Name' field. Customize it to your needs.",
```
Comment (pyphilia) [inline src/langs/en.json:32]:
```suggestion
  "CREATE_NEW_ITEM_DISPLAY_NAME_HELPER_TEXT": "Show in Graasp Player. Defaults to the 'Name' field. Customize it to your needs.",
```

**Review by pyphilia — APPROVED** [review]
All good to me, you just need to rebase and we can merge this branch.

File: src/components/item/form/DocumentForm.tsx [hunk 6]
```diff
@@ -188,12 +286,19 @@ const DocumentForm = ({
...
-        updatedProperties={updatedProperties}
-      />
+      <Stack direction="row" spacing={2}>
+        <NameForm
+          setChanges={setChanges}
+          item={item}
+          required
+          updatedProperties={updatedProperties}
+        />
+        <DisplayNameForm
```
Comment (pyphilia) [inline src/components/item/form/DocumentForm.tsx:296]: Actually the display name is on autoFocus. I think it's the `name` field that should be focused by default.

**Review by pyphilia — CHANGES_REQUESTED** [review]

**Comment by ztlee042** [comment]
> All good to me, you just need to rebase and we can merge this branch.

I rebased, hope I get it right :)

**Review by pyphilia — APPROVED** [review]

**Pull request merged**